### DISCUSSION TITLE
Why populateExchange doesn't work with interface/union?

### DISCUSSION THREAD

**Discussion by nghiepdev** [urql-graphql/urql#909] [Q&A] [answered]
Hi all,

```
Graphcache Error: Invalid TypeInfo state: Found no flat schema type when one was expected
```

**My schema example:**
```graphql
interface Product {
  id: ID!
  name: String!
  sku: String!
  price: UnsignedInt!
}

type SimpleProduct implements Product {
  id: ID!
  name: String!
  sku: String!
  price: UnsignedInt!
}

type ComplexProduct implements Product {
  id: ID!
  name: String!
  sku: String!
  price: UnsignedInt!
  tax: UnsignedInt!
  fee: UnsignedInt!
}
``` 

Doesn't work with an interface and throw an error same the above:
```graphql
type Query {
  products: [Product]!
}
```

Work with separate multiple query:
```graphql
type Query {
  simpleProducts: [SimpleProduct]!
  complexProducts: [ComplexProduct]!
}
```

**Comment by kitten** [comment]
I think you mean the populate exchange? And you haven't posted more examples here so I have to make an assumption.

The `populateExchange` is still not quite stable and complete, so it doesn't support any edge cases around interfaces and unions. In fact, the error you see is the error you get with those.

But if we figure out how to handle these edge cases correctly, we'll push out a new version ✌️

**Comment by JoviDeCroock** [comment, marked as answer]
This should be fixed in the latest populate-exchange version which is 0.2.0

**Reply by nghiepdev** [reply]
Thanks @JoviDeCroock
